### DISCUSSION TITLE
Customizing Tina UI

### DISCUSSION THREAD

**Discussion by KevinLanvin** [tinacms/tinacms#4405] [Help]
Hello,
I'm trying to use Tina with Astro as SSG, and I wanted to add a button to UI to trigger a webhook for online site building.
I couldn't find a way to add such button in sidebar for example. Is it even possible to add custom components to Tina screen ? If not, is it a feature that is going to be introduced  ?

**Comment by arunavo4** [comment]
When you save a file in Tina cms it should trigger a commit and when that happens if you have a CI/CD Pipeline setup then you would automatically build the site and deploy it.

**Reply by KevinLanvin** [reply]
Yes this feature is working fine but I wanted more control on build triggers. I wanted to only rebuild the site after all modifications are pushed, not rebuild on each push. So I wanted to use a webhook to trigger the build but cannot bind it to a custon button in Tina interface

**Reply by arunavo4** [reply]
I think what you are referring to is the editorial workflow that Tina offers so you can use PR for all the changes and then later merge the branch but they have restricted that to cloud users 

https://tina.io/docs/drafts/editorial-workflow/


It would have been nice to have an alternative to this in the open source version. I will try and investigate if we can achieve this feature some how.

**Reply by kldavis4** [reply]
I think for self-hosted editorial workflow, you'd need to provide the branch related endpoints being called here: https://github.com/tinacms/tinacms/blame/main/packages/tinacms/src/internalClient/index.ts

@KevinLanvin I don't think we currently provide a clean way for updating the UI - though you may want to jump on discord and search or post in #help as I believe this has been requested before.
